### PR TITLE
contract-types: Use Node 20 tsconfig preset

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Review comment

### CODE REVIEW

File: packages/contract-types/dist/contract-types/src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * as Contracts from "./generated";
```
Comment (nalinbhardwaj) [inline packages/contract-types/dist/contract-types/src/index.d.ts:1]: @dcposch do you know if there's a way to not commit these dist files but build them for other package includes? do we just need to add a publish pipeline to npm or something for that?

I think some yarn/npm workspace magic should also be able to do this automagically, but I can't find on googling easily

File: packages/contract-types/package.json [hunk 2]
```diff
@@ -2,17 +2,18 @@
   "name": "daimo-contract-types",
   "version": "1.0.0",
   "description": "Typescript types and ABIs for daimo contracts",
-  "main": "src/index.ts",
+  "types": "./dist/contract-types/src/index.d.ts",
+  "main": "./dist/contract-types/src/index.js",
```
Comment (dcposch) [inline packages/contract-types/package.json:6]: was something breaking linking to `src/index.ts`?

Building typescript is generally necessary when publishing a library to NPM, but if we're only using it locally from another typescript project, I'm not sure

Comment (nalinbhardwaj) [reply packages/contract-types/package.json:6]: yeah https://github.com/daimo-eth/daimo/pull/25#discussion_r1225900978 requires us to `npm pack` these which only picks from the built files.

**Comment by dcposch** [comment]
combined into #26

**Pull request closed without merging**